src/createHeadersInterceptor.ts

```diff
@@ -0,0 +1,10 @@
+import type { Interceptor } from "@bufbuild/connect";
+
+export function createHeadersInterceptor(headers: any): Interceptor {
```

Comment:
Move method to `@substreams/core`

Perhaps in `/auth` folder? https://github.com/substreams-js/substreams-js/tree/main/packages/core/src/auth

Or `/utils`

https://github.com/substreams-js/substreams-js/tree/main/packages/core/src/utils

@fubhy can chime in for a preference